packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -134,17 +137,13 @@ NS_INLINE CGFloat radiansToDegrees(CGFloat radians) {
   return degrees;
 };
 
-NS_INLINE UIViewController *rootViewController() API_AVAILABLE(ios(16.0)) {
-  for (UIScene *scene in UIApplication.sharedApplication.connectedScenes) {
-    if ([scene isKindOfClass:UIWindowScene.class]) {
-      for (UIWindow *window in ((UIWindowScene *)scene).windows) {
-        if (window.isKeyWindow) {
-          return window.rootViewController;
-        }
-      }
-    }
-  }
-  return nil;
+NS_INLINE UIViewController *rootViewController() {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  // TODO: (hellohuanlin) Provide a non-deprecated codepath. See
+  // https://github.com/flutter/flutter/issues/104117
+  return UIApplication.sharedApplication.keyWindow.rootViewController;
+#pragma clang diagnostic pop
```

Comment:
I simply copied the code from another plugin. Fixing it is not the scope of this PR (tho I'm happy to help if needed).